perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -388,60 +446,55 @@ def Run(benchmark_spec):
+                                               FLAGS.io_depths,
+                                               FLAGS.working_set_size)
+
+  samples = []
+
+  def RunIt(repeat_number=None, minutes_since_start=None, total_repeats=None):
+    if repeat_number:
+      logging.info('**** Repetition number %s of %s ****',
+                   repeat_number, total_repeats)
 
-  if not FLAGS.generate_scenarios:
     stdout, stderr = vm.RemoteCommand(fio_command, should_log=True)
 
-    return fio.ParseResults(
-        GetOrGenerateJobFileString(FLAGS.fio_jobfile,
-                                   disk,
-                                   FLAGS.against_device,
-                                   FLAGS.generate_scenarios,
-                                   GetIODepths(FLAGS.io_depths),
-                                   FLAGS.working_set_size),
-        json.loads(stdout))
-  else:
-    # We want to run each scenario for FLAGS.run_for_minutes time.
-    run_reps = FLAGS.run_for_minutes // MINUTES_PER_JOB
-    if FLAGS.run_for_minutes % MINUTES_PER_JOB != 0:
-      run_reps += 1
-      # We already warned the user about rounding during the prepare
-      # phase. No need to warn them again here.
-
-    samples = []
-    start_time = datetime.datetime.now()
-    for rep_num in xrange(run_reps):
-      logging.info('**** Repetition number %s of %s ****', rep_num, run_reps)
-      run_start = datetime.datetime.now()
-      stdout, stderr = vm.RemoteCommand(fio_command, should_log=True)
-
-      seconds_since_start = int(round((run_start - start_time).total_seconds()))
-      minutes_since_start = int(round(float(seconds_since_start)
-                                      / float(SECONDS_PER_MINUTE)))
+    if repeat_number:
       base_metadata = {
-          'repeat_number': rep_num,
+          'repeat_number': repeat_number,
           'minutes_since_start': minutes_since_start
       }
-      samples.extend(fio.ParseResults(
-          GetOrGenerateJobFileString(FLAGS.fio_jobfile,
-                                     disk,
-                                     FLAGS.against_device,
-                                     FLAGS.generate_scenarios,
-                                     GetIODepths(FLAGS.io_depths),
-                                     FLAGS.working_set_size),
-          json.loads(stdout),
-          base_metadata=base_metadata))
-
-    return samples
+    else:
+      base_metadata = None
+
+    samples.extend(fio.ParseResults(job_file_string,
+                                    json.loads(stdout),
+                                    base_metadata=base_metadata))
+
+  # TODO(user): This only gives results at the end of a job run
+  #      so the program pauses here with no feedback to the user.
+  #      This is a pretty lousy experience.
+  logging.info('FIO Results:')
+
+  if not FLAGS.run_for_minutes:
+    RunIt()
+  else:
+    RunForMinutes(RunIt, FLAGS.run_for_minutes, MINUTES_PER_JOB)
```

Comment:
If you use a custom job flie and pass run_for_minutes, then it will print a warning and ignore your run_for_minutes flag. This is because run_for_minutes is in FLAGS_IGNORED_FOR_CUSTOM_JOBFILE at the top of the file.